### PR TITLE
Change Replicated upgrade policy to Online

### PR DESCRIPTION
**Pull request by roypaulin**
The new upgrade policy will be `Online` instead of `Replicated`. What was formerly called `Online` is now `ReadOnlyOnline` upgrade

### CODE REVIEW

**Comment by cchen-vertica** [comment]
Even when online-upgrade(replicated upgrade) is robust, I think we still need to support read-only online-upgrade because it's much faster (no sandbox needed) and cheaper (no extra pods needed). If this is the case, read-only online-upgrade will be another policy type.

**Comment by roypaulin** [comment]
> Even when online-upgrade(replicated upgrade) is robust, I think we still need to support read-only online-upgrade because it's much faster (no sandbox needed) and cheaper (no extra pods needed). If this is the case, read-only online-upgrade will be another policy type.

That's probably better. We need to document that change.

File: api/v1/verticadb_types.go [hunk 1]
```diff
@@ -128,17 +128,17 @@ type VerticaDBSpec struct {
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:select:Auto","urn:alm:descriptor:com.tectonic.ui:select:Online","urn:alm:descriptor:com.tectonic.ui:select:Offline"}
 	// +kubebuilder:default:=Auto
 	// This setting defines how the upgrade process will be managed. The
-	// available values are Offline, Online, Replicated, and Auto.
+	// available values are Offline, Online, and Auto.
```
Comment (cchen-vertica) [inline api/v1/verticadb_types.go:131]: Add ReadOnlyOnline here

File: api/v1/verticadb_types.go [hunk 2]
```diff
@@ -416,12 +416,14 @@ const (
 	// leaving the secondary subclusters in read-only mode.  When the primary
 	// subcluster comes back up, we restart/remove all of the secondary
 	// subclusters to take them out of read-only mode.
-	OnlineUpgrade UpgradePolicyType = "Online"
-	// Like online upgrade, however it allows for writes. This is done by
+	// This is only used internally by the operator to differentiate between
```
Comment (cchen-vertica) [inline api/v1/verticadb_types.go:419]: ReadOnlyOnlineUpgrade shouldn't be used internally.

File: config/manifests/bases/verticadb-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -227,6 +227,14 @@ spec:
         path: template.isPrimary
         x-descriptors:
         - urn:alm:descriptor:com.tectonic.ui:booleanSwitch
+      - description: A sandbox primary subcluster is a secondary subcluster that was
```
Comment (cchen-vertica) [inline config/manifests/bases/verticadb-operator.clusterserviceversion.yaml:230]: Thanks for adding this.

File: tests/e2e-leg-9/new-online-upgrade-sanity/co-assert.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
+# (c) Copyright [2021-2024] Open Text.
```
Comment (cchen-vertica) [inline tests/e2e-leg-9/new-online-upgrade-sanity/co-assert.yaml:1]: What is this for? When will it be verified?

File: tests/e2e-leg-9/new-online-upgrade-sanity/co-wait-for-sc-promotion.yaml [hunk 5]
```diff
@@ -0,0 +1 @@
+# Intentionally empty to give this step a name in kuttl
```
Comment (cchen-vertica) [inline tests/e2e-leg-9/new-online-upgrade-sanity/co-wait-for-sc-promotion.yaml:1]: Ditto

**Comment by cchen-vertica** [comment]
Looks good overall. I only have some minor comments.

File: api/v1/verticadb_types.go (hunk 2 above)
Comment (roypaulin) [reply api/v1/verticadb_types.go:419]: That's an old comment. Not relevant anymore, will update it

File: tests/e2e-leg-9/new-online-upgrade-sanity/co-assert.yaml (hunk 4 above)
Comment (roypaulin) [reply tests/e2e-leg-9/new-online-upgrade-sanity/co-assert.yaml:1]: Should have ever been committed. Will remove it.

**Review by cchen-vertica — APPROVED** [review]

**Pull request merged**